joomla/joomla-cms
# [4.0] Rename Fields - Subfields plugin to Fields - Subform plugin

Pull Request for Issue # .

### Summary of Changes
With PR https://github.com/joomla/joomla-cms/pull/32864,  @brianteeman made some changes to language files to change subfields plugin to subform plugin. However, the actually plugin code is still uses old name **subfields**. This PR changes the plugin code to have it completely renamed from **subfields** to **subform** 

### Testing Instructions
1. Install Joomla 4 nightly build. Or if you have Joomla 4 installation already, please update that installation to Joomla 4 nightly build
2. Access to Content -> Fields, create at least one **subfields** custom field. For testing purpose, maybe you can create two text custom fields (these fields should have Categories set to None) and one subfields like in the attached screenshot (the subfields could have two text fields in it's Fields setting)

- List of Fields
![subfields](https://user-images.githubusercontent.com/977664/115417889-36178c00-a223-11eb-8b9c-14fcfb1ad4af.png)

- Subfields settings

![subfields_setting](https://user-images.githubusercontent.com/977664/115418164-78d96400-a223-11eb-8f12-fb789c03ec1f.png)

3. Create a new article, go to Fields tab and enter data for that subfields custom field for the article

4. Download update package for this PR at https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33200/downloads/42161/Joomla_4.0.0-beta8-dev+pr.33200-Development-Update_Package.zip , go to System -> Update -> Joomla, install that update package

5. Now, access to Content -> Fields again, confirm that the **subfields** which you created before is now become **subform**

![subfields_become_subform](https://user-images.githubusercontent.com/977664/115419001-2b112b80-a224-11eb-8765-8a09e5f24d49.png)

6. Edit that subform custom field (before it was subfields), confirm that **Type** now become **Subform (Subform)**. Also, the Fields setting still same as before
![type_changed](https://user-images.githubusercontent.com/977664/115419262-64499b80-a224-11eb-9e75-c5674e1bca4b.png)

7. Edit the article, go to Fields tab, confirm that the data which you entered for the custom fields before is still there.

8. View the article from frontend (maybe make it featured so that it is displayed at homepage by default) and confirm that the data you entered for the custom fields is being displayed properly

![article](https://user-images.githubusercontent.com/977664/115419662-b5598f80-a224-11eb-9f81-e844b7914c40.png)
